tests/unit/app/endpoints/test_ols.py

```diff
@@ -36,6 +36,50 @@ def test_retrieve_conversation_id_existing_id(load_config):
     assert new_id == old_id, "Old (existing) ID should be retrieved." ""
 
 
+@patch("ols.utils.config.conversation_cache.insert_or_append")
+def test_store_conversation_history(insert_or_append, load_config):
+    """Test if operation to store conversation history to cache is called."""
+    user_id = "1234"
+    conversation_id = suid.get_suid()
+    query = "Tell me about Kubernetes"
+    llm_request = LLMRequest(query=query)
+    response = ""
+    ols.store_conversation_history(user_id, conversation_id, llm_request, response)
+    insert_or_append.assert_called_with(user_id, conversation_id, f"{query}\n\n")
+
+
+@patch("ols.utils.config.conversation_cache.insert_or_append")
+def test_store_conversation_history_some_response(insert_or_append, load_config):
+    """Test if operation to store conversation history to cache is called."""
+    user_id = "1234"
+    conversation_id = suid.get_suid()
+    query = "Tell me about Kubernetes"
+    llm_request = LLMRequest(query=query)
+    response = "*response*"
+    ols.store_conversation_history(user_id, conversation_id, llm_request, response)
+    insert_or_append.assert_called_with(
+        user_id, conversation_id, f"{query}\n\n{response}"
+    )
+
+
+def test_store_conversation_history_improper_user_id(load_config):
+    """Test if basic input verification is done during history store operation."""
+    user_id = "::::"
+    conversation_id = suid.get_suid()
+    llm_request = LLMRequest(query="Tell me about Kubernetes")
+    with pytest.raises(ValueError, match="Incorrect user ID"):
+        ols.store_conversation_history(user_id, conversation_id, llm_request, "")
+
+
+def test_store_conversation_history_improper_conversation_id(load_config):
+    """Test if basic input verification is done during history store operation."""
+    user_id = "1234"
+    conversation_id = "::::"
+    llm_request = LLMRequest(query="Tell me about Kubernetes")
+    with pytest.raises(ValueError, match="Incorrect conversation ID"):
```

Comment:
"invalid", not "incorrect" (need to fix the error message)

Reply:
yup, we'll first need to merge https://github.com/openshift/lightspeed-service/pull/407  then I'll update this one

Reply:
fixed @bparees